File: packages/sdkwork-utils-python/sdkwork_utils/boolean.py
from sdkwork_utils.string import is_blank


def parse_bool(value: str) -> bool | None:
    match value.strip().lower():
        case "true" | "1" | "yes" | "on":
            return True
        case "false" | "0" | "no" | "off":
            return False
        case _:
            return None


def is_truthy(value: str | None) -> bool:
    if is_blank(value):
        return False
    return value.strip().lower() not in {"false", "0", "no", "off"}

File: packages/sdkwork-utils-typescript/src/boolean.ts
import { isBlank } from "./string.js";

export function parseBool(value: string): boolean | null {
  switch (value.trim().toLowerCase()) {
    case "true":
    case "1":
    case "yes":
    case "on":
      return true;
    case "false":
    case "0":
    case "no":
    case "off":
      return false;
    default:
      return null;
  }
}

export function isTruthy(value: string | null | undefined): boolean {
  if (isBlank(value)) {
    return false;
  }
  switch (value!.trim().toLowerCase()) {
    case "false":
    case "0":
    case "no":
    case "off":
      return false;
    default:
      return true;
  }
}
